packages/framework/src/client/scheduler.js

```diff
@@ -0,0 +1,203 @@
+		} );
+	}
+
+	/**
+	 * Schedule a read of a resource name according to the requirement set given.
+	 * @param {string} resourceName The name of the resource to be read
+	 * @param {Object} requirement The set of requirements (freshness, timeout)
+	 * @param {Object} resourceState The current snapshot of resource state
+	 * @param {Date} now The current time.
+	 */
+	scheduleRequest = ( resourceName, requirement, resourceState, now = new Date() ) => {
+		if ( this.getInFlightRequests( resourceName, now ).length > 0 ) {
+			// Do nothing, there's already a request in flight.
+			return;
+		}
+
+		const existingRequest = this.getScheduledRequest( resourceName, now );
+		if ( existingRequest ) {
+			existingRequest.addRequirement( requirement, resourceState, now );
+		} else {
+			this.requests.push( new ResourceRequest( resourceName, requirement, resourceState, now ) );
+		}
+		this.updateDelay( now );
+	};
+
+	/**
+	 * Send a list of requests.
+	 * @param {Array} requests The list of requests to send.
+	 * @param {Date} now The current time.
+	 * @return {Promise} A promise returned from the fetch call.
+	 */
+	sendRequests = ( requests, now ) => {
+		if ( requests.length > 0 ) {
+			const promise = this.fetch( requests.map( request => request.resourceName ) );
+
+			requests.forEach( ( request ) => {
+				request.requested( promise, now );
+			} );
+			return promise;
+		}
+		// No requests to send.
+		return Promise.resolve();
+	};
+
+	/**
+	 * Send any scheduled requests that are ready to be sent.
+	 * @param {Date} now The current time.
+	 * @return {Promise} A promise returned from the fetch call.
+	 */
+	sendReadyRequests = ( now = new Date() ) => {
+		const readyRequests = this.requests.filter( ( request ) => {
+			return this.isRequestReady( request, now );
+		} );
+		return this.sendRequests( readyRequests, now );
+	};
+
+	/**
+	 * Send any scheduled requests that are timed out.
+	 * @param {Date} now The current time.
+	 * @return {Promise} A promise returned from the fetch call.
+	 */
+	resendTimeouts = ( now = new Date() ) => {
+		const timedOutRequests = this.requests.filter( ( request ) => {
+			return STATUS.timedOut === request.getStatus( now );
+		} );
+		return this.sendRequests( timedOutRequests, now );
+	};
+
+	/**
+	 * Clear out completed and failed requests.
+	 * @param {Date} now The current time.
+	 */
+	cleanUp = ( now = new Date() ) => {
+		this.requests = this.requests.filter( ( request ) => {
+			const status = request.getStatus( now );
+			return STATUS.complete !== status && STATUS.failed !== status;
+		} );
+	};
+
+	/**
+	 * Checks if a request is ready.
+	 * @param {ResourceRequest} request The request to check.
+	 * @param {Date} now The current time.
+	 * @return {boolean} True if the request is ready to be sent, false otherwise.
+	 */
+	isRequestReady = ( request, now = new Date() ) => {
+		const status = request.getStatus( now = new Date() );
```

Comment:
```suggestion
		const status = request.getStatus( now );
```

Reply:
Fixed in f44b57c0276d2b7125e701eed713f9f35dd57f55